docs/git/branches.md

```diff
@@ -0,0 +1,169 @@
+
+TODO: Image of commit history
+
+Since then, you've been pressing a button in GitHub Desktop called "Commit to main" to make a new commit. But what does that "to main" mean? main is called a _branch_, which is a list of commits. It turns out, you can have more than one branch, and they can share some commits if they want:
+
+TODO: Image of two divergent branches with shared history, highlighting which is shared and which is divergent.
+
+### Why would we want to do this?
+
+There are a few reasons. The main one is: So you can work on more than one thing at the same time.
+
+Let's look at our repo of recipes. Right now, it contains one recipe, for a brownie in a mug. There are a couple of changes we want to make:
+
+1. We want to add an optional topping to put on top of the brownie (maybe ice cream, maybe you have a better idea). We'll need to add it both as an extra ingredient, and an extra step at the end.
+2. A lot of recipes start with a story of how the author found them, or some time they've really enjoyed the recipe. We want to add a story, before the "What you need" section.
+
+Let's imagine we start writing the story, but we want to quickly add the optional ice cream step. And let's imagine that we want someone else to review both changes before we merge them to main.
+
+We could make both changes to main, and ask someone to review them both, but then they both need to be right before we can merge either one - we can't merge just the ice cream step until the story is finished.
+
+Or we could only make the ice cream change, then send it for review, then when that merges, write the story and send it for review. But that slows us down, and also means that the reviewer can't review both changes at the same time.
+
+It would be better if we could send out one pull request for the ice cream change, and another for the story. We can do this - with branches:
+
+TODO: Same image as before, of divergent branches, but labelled "ice cream" and "story".
+
+We can then add new commits on top of each branch, separately, and later on merge both branches to main.
+
+### How do we create a new branch?
+
+```console
+$ git switch --create story
+Switched to a new branch 'story'
+```
+
+Let's run through the bits of that command:
+
+`git` - we're running a git command.
+
+`switch` - we're switching to a different branch than the one we're on right now.
+
+`--create` - we want git to create the branch we're switching to.
+
+`story` - the branch we want to switch to (and create) is called `story`.
+
+Running this command doesn't change any of the files in our repository - they stay the same, but if we commit changes, they will only get committed on this branch. If we switched back to the `main` branch, the changes would be un-done. But if we switch back to `story`, they'll come back!
+
+### How do I know what branch I'm on?
+
+You've already run one command which tells you this, you just may not have noticed!
```

Comment:
Have we already run this?

Reply:
I'm hoping `git status` will come up in the intro to git cli class, but may need to edit this if not :)